### ISSUE TITLE
Make the bot and code more server agnostic

### ISSUE THREAD

**Issue by ESloman** [ESloman/bsebot#79]
The bot has a lot of functionality that's dependent on channels/roles within the BSE server.
This should change and allow the bot to operate better - regardless of which server it's running in.
Things that rely on specific channels/roles:
- wordle stuff
- 'the boys' role for eddie distribution
- needing a 'bseddies-chat' channel for posting KING stuff, update messages, etc
- the need for a 'bseddies KING' role

Some of this could be mitigated - having the bot create these roles and channels when it joins and storing the IDs of them in the DB so it's easily accessible to the bot (rather than relying on hardcoded values).

There's no real rush - not particularly interested in making this publicly available but it would be good to follow best practices.

**Comment by ESloman** [comment]
Will need a particular `guilds` collection; where every document is a server the bot is in. We can then grab the `bseddies` channel ID from there and the `bseddies KING` role ID as well. If we don't have a document for a server, or we're missing the channel/role, then we should create it. We should also create a `bseddies` category for the channel.

**Comment by ESloman** [comment]
DB doc thoughts:

```python
{
    "guild_id": GUILD_ID,
    "category": CATEGORY_ID,
    "channel": CHANNEL_ID,
    "role": ROLE_ID,
    "wordle_channel": WORDLE_CHANNEL,  # could not exist or be None - for guilds that don't want wordle automatically
    "jerk_channel": JERK_OFF_CHANNEL_ID  # mostly for BSE server - not relevant for others
}

```

**Comment by ESloman** [comment]
Begun implementing this slowly - the Collection now exists and the slash command implemented in #206 relies on that for functionality. Some of this stuff will need populating manually - need to add a slash command to configure it.

**Comment by ESloman** [comment]
Some other stuff that needs changing:
- a lot of the slash commands use a `if ctx.guild.id not in self.guilds: return` type check; so they only work in servers they've been told to work in. `BSEBot`, won't respond to commands even though it's been invited to the sloman server.

**Comment by ESloman** [comment]
Further changes made to the Guilds collection here: https://github.com/ESloman/bsebot/pull/274

**Comment by ESloman** [comment]
This was resolved as #250 was resolved. The bot is pretty server independent now and there's significantly less hard-coded values. There's still _some_ but they'll continue to be removed in the future.
